announce_test.go

```diff
@@ -113,12 +113,23 @@ func TestAnnounceReplace(t *testing.T) {
 		t.Fatal(err)
 	}
 	t.Log("Sent announce for last CID", lastCid)
-	// Check that the pending CID is set to the last one announced.
-	hnd.qlock.Lock()
-	pendingCid = hnd.pendingCid
-	hnd.qlock.Unlock()
+
+	// Check that the pending CID is set to the last one announced. Need to
+	// check in retry loop because there is no signal to indicate when the
+	// announce handler goroutine delivered the announce to Subscriber for
+	// handling.
+	for i := 0; i < 10; i++ {
+		hnd.qlock.Lock()
+		pendingCid = hnd.pendingCid
+		hnd.qlock.Unlock()
+		if pendingCid == lastCid {
+			break
+		}
+		time.Sleep(10 * time.Millisecond)
```

Comment:
what if you miss the pendingCid? is it possible for it to get over-written in a 10ms window?

Reply:
Fixed by exiting the loop only when the CID stops changing for long enough to be certain it is the last value that will be seen. Then that last value is checked against the expected.